File: src/main/java/uk/co/squadlist/web/urls/SeoLinkBuilder.java
package uk.co.squadlist.web.urls;

import org.springframework.stereotype.Component;

@Component
public class SeoLinkBuilder {

	public String makeSeoLinkFor(String name) {
		String result = name.toLowerCase().trim().replaceAll("\\s", "-");
		result = result.replaceAll("[^\\-a-z0-9_]", "");
		result = result.replaceAll("--+", "-");
		return result;
	}
	
}

File: src/main/kotlin/uk/co/squadlist/web/urls/SeoLinkBuilder.kt
package uk.co.squadlist.web.urls

class SeoLinkBuilder {

    fun makeSeoLinkFor(name: String): String {
        var result = name.toLowerCase().trim { it <= ' ' }.replace("\\s".toRegex(), "-")
        result = result.replace("[^\\-a-z0-9_]".toRegex(), "")
        result = result.replace("--+".toRegex(), "-")
        return result
    }

}
